docarray/doc_index/abstract_doc_index.py

```diff
@@ -458,7 +484,15 @@ def filter(
         :param limit: maximum number of documents to return
         :return: a DocumentArray containing the documents that match the filter query
         """
-        return self._filter(filter_query, limit=limit, **kwargs)  # type: ignore
+        docs = self._filter(filter_query, limit=limit, **kwargs)
+
+        if isinstance(docs, List):
+            da_cls = DocumentArray.__class_getitem__(
+                cast(Type[BaseDocument], self._schema)
+            )
+            docs = da_cls(self._convert_to_doc_list(docs))
```

Comment:
It might be worth breaking this logic into a separate private method, something like `_list_to_docarray()`.